### ISSUE TITLE
Regression for truthy bool with getattr

### ISSUE THREAD

**Issue by ilinum** [python/mypy#14480]
The following code generates an error on latest master but no on latest release (with `--enable-error-codes=truthy-bool`):
```python
def f(obj: object) -> None:
    print("{}".format(
        getattr(type(obj), "__foo__", "") or "unknown"  # E: "builtins.getattr" returns "object" which does not implement __bool__ or __len__ so it could always be true in boolean context  [truthy-bool]
    ))

f("")
```

It's clearly a false positive, if your run the code you see that it prints `unknown`.

**Comment by JukkaL** [comment]
Here's a bit simpler example that reproduces the issue:
```py
def f(t: type[object]) -> None:
    o: object = getattr(t, "x", "") or ""
```
This looks different from #14481, as there are no TypedDicts involved here.
